res/controllers/DJControl-Compact-scripts.js

```diff
@@ -0,0 +1,113 @@
+// Mode button does give out midi, but the selection itself is done inside the
+// controller -- no control over which mode is selected for instance.
+
+HercDJCompact = new function() {
+   this.group = "[Master]";
+}
+
+HercDJCompact.init = function(id) {
+    scratch = false;
+    scratch_timer = [];
+    scratch_timer_on = [];
+
+    engine.connectControl("[Recording]", "status",
+            "HercDJCompact.OnRecordingStatusChange");
+
+    // Tell controller to send midi to update knob and slider positions.
+    midi.sendShortMsg(0xB0, 0x7F, 0x7F);
```

Comment:
I tried this with my Electrix Tweaker just to see if it would work but it didn't. Is this a Hercules thing? I think I recall looking at one other mapping that had such a feature, but I don't remember what mapping it was or what MIDI signal was sent.